tests/examples/CMakeLists.txt

```diff
@@ -14,22 +14,44 @@ cmake_minimum_required(VERSION 3.0.0)
 # Get and update the LUA_PATH so the scripts can be executed without Awesome.
 execute_process(COMMAND lua -e print\(package.path\) OUTPUT_VARIABLE "LUA_PATH_")
 
+# Make sure the system can be called from the test directory
+if(NOT SOURCE_DIR AND ${CMAKE_CURRENT_SOURCE_DIR} MATCHES "/tests/examples")
+    set(SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../")
+
+    # .luacov use these variables
+    set(ENV{BUILD_DIRECTORY} ${CMAKE_CURRENT_BINARY_DIR}/)
+    set(ENV{AWESOME_LIB_DIR} ${SOURCE_DIR}/lib/)
+    set(ENV{SOURCE_DIRECTORY} ${SOURCE_DIR})
+else()
+    set(SOURCE_DIR ${CMAKE_SOURCE_DIR})
+endif()
```

Comment:
"if `SOURCE_DIR` is set, overwrite it". That seems odd to me. How about giving this variable a different name instead (dunno, `TOP_SOURCE_DIR`)?